src/sentry/static/sentry/app/views/onboarding/wizardNew.jsx

```diff
@@ -0,0 +1,181 @@
+import InlineSvg from 'app/components/inlineSvg';
+import OnboardingPlatform from 'app/views/onboarding/platform';
+import OnboardingProjectSetup from 'app/views/onboarding/projectSetup';
+import OnboardingWelcome from 'app/views/onboarding/welcome';
+import PageHeading from 'app/components/pageHeading';
+import SentryTypes from 'app/sentryTypes';
+import space from 'app/styles/space';
+import withProjects from 'app/utils/withProjects';
+
+const ONBOARDING_STEPS = [
+  {
+    id: 'welcome',
+    title: t('Welcome to Sentry'),
+    component: OnboardingWelcome,
+  },
+  {
+    id: 'select-platform',
+    title: t('Platform Selection'),
+    component: OnboardingPlatform,
+  },
+  {
+    id: 'get-started',
+    title: t('Getting Started with Sentry'),
+    component: OnboardingProjectSetup,
+  },
+];
+
+class OnboardingWizard extends React.Component {
+  static propTypes = {
+    projects: PropTypes.arrayOf(SentryTypes.Project),
+  };
+
+  state = {};
+
+  get activeStepIndex() {
+    return ONBOARDING_STEPS.findIndex(({id}) => this.props.params.step === id);
+  }
+
+  get activeStep() {
+    return ONBOARDING_STEPS[this.activeStepIndex];
+  }
+
+  get firstProject() {
+    const sortedProjects = this.props.projects.sort(
+      (a, b) => new Date(b.dateCreated) - new Date(a.dateCreated)
+    );
+
+    return sortedProjects.length > 0 ? sortedProjects[0] : null;
+  }
+
+  get projectPlatform() {
+    return this.state.platform || this.firstProject?.platform;
+  }
+
+  handleUpdate = data => {
+    this.setState(data);
+  };
+
+  handleNextStep = step => data => {
+    this.handleUpdate(data);
+
+    if (step !== this.activeStep) {
+      return;
+    }
+
+    const {orgId} = this.props.params;
+    const nextStep = ONBOARDING_STEPS[this.activeStepIndex + 1];
+    browserHistory.push(`/onboarding-new/${orgId}/${nextStep.id}/`);
+  };
+
+  handleReturnToStep = step => data => {
+    const {orgId} = this.props.params;
+
+    this.handleUpdate(data);
+    browserHistory.push(`/onboarding-new/${orgId}/${step.id}/`);
+  };
+
+  scrollToActiveStep = () => {
+    const step = this.activeStep;
+    scrollToElemennt(`#onboarding_step_${step.id}`, {
+      align: 'middle',
+      duration: 300,
+    });
+  };
+
+  renderOnboardingSteps() {
+    const activeStepIndex = this.activeStepIndex;
+    const {orgId} = this.props.params;
+
+    if (activeStepIndex === -1) {
+      // TODO: Redirect here?
+    }
+
+    const visibleSteps = ONBOARDING_STEPS.slice(0, activeStepIndex + 1);
+
+    return visibleSteps.map((step, index) => (
+      <OnboardingStep key={step.id} onPoseComplete={this.scrollToActiveStep}>
+        <PageHeading withMargins>{step.title}</PageHeading>
+        <step.component
+          scrollTargetId={`onboarding_step_${step.id}`}
+          active={activeStepIndex === index}
+          orgId={orgId}
+          project={this.firstProject}
+          platform={this.projectPlatform}
+          onReturnToStep={this.handleReturnToStep(step)}
```

Comment:
Kind of a nit, but I can see how this can potentially confuse people in the future because we don't otherwise use this pattern.

Can we instead, inline the closure here (or use `bind()`) instead of having the handler method return a closure?

Reply:
135fefa